### DISCUSSION TITLE
How to use ConfigMap with podman-kube@.service?

### DISCUSSION THREAD

**Discussion by sid-6581** [containers/podman#16538] [Q&A]
With `podman play kube` we can use `--configmap` to pass `ConfigMap` files, but the service file is hardcoded with just the arguments ` --replace --service-container=true`. Do we just need to put everything in one file?

**Comment by rhatdan** [comment]
@umohnani8 PTAL

**Comment by umohnani8** [comment]
Hi, looks like you are trying to move form a kube yaml to a systemd service file, correct? It does seem like we don't have a way of pointing to a configMap for that workflow like we have for `podman play kube`.
One thing you can try is putting the configMap in the same yaml file as your workload definition and seeing if the systemd service file created has that. If it doesn't work, please open an issue with some details and example and we will work on adding the missing support.

**Reply by sid-6581** [reply]
That's correct! Well, technically we're trying to move from `podman-compose` with its built-in systemd service files directly to `podman` kube systemd services. With `podman-compose` we're using `.env` files, but it seems like there's nothing analogous here. That's not a big deal since everything is scripted and templated, and we can easily add a configMap document on top of the pod definition in the same file.

I'm planning on trying this on a couple of services this weekend, so I'll report back on how that goes.
